core/includes/file.inc

```diff
@@ -2018,6 +2018,64 @@ function file_validate_image_orientation(File $file, $orientation = 0) {
   return $errors;
 }
 
+/**
+ * Validate uploaded SVG files.
+ *
+ * @param File $file
+ *   A file entity.
+ *
+ * @return
+ *   An array. If the file is an SVG and is either not valid or contains
+ *   dangerous content, it will contain an error message.
+ *
+ * @see hook_file_validate()
+ */
+function file_validate_svg(File $file) {
+  $errors = array();
+
+  if (image_is_svg($file->uri)) {
+    $file_contents = file_get_contents($file->uri);
+    $xml = simplexml_load_string($file_contents);
```

Comment:
OK, we have to wrap this in libxml_use_internal_errors() to capture errors (as in my example).

Reply:
> we have to wrap this in libxml_use_internal_errors() to capture errors 

It wasntnecessary to do so on my local machine. The reason the upload validation wasnt working was actually the `if (image_is_svg($file->uri))` check was failing. Fixed it, but no sandbox yet to test (need rebase again!).

Reply:
I wrapped `libxml_use_internal_errors()` and it's capturing invalid svg and returning "Invalid SVG file." There's nothing in the logs so I removed that. We can have a TODO to output the svg errors to the screen in a friendly way.

It would be good to have a test with an invalid svg and a dirty svg.

I noticed that this function isn't called if creating a file from `admin/content/files`.

Reply:
I fixed it so it gets called for file uploads too.